### PR TITLE
SDIT-1826 Add endpoint to upsert physical attributes

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderPhysicalAttributes.kt [hunk 1]
```diff
@@ -66,20 +65,4 @@ class OffenderPhysicalAttributes(
...
-      heightFeet?.let { ((it * 12) + (heightInches ?: 0)) * 2.54 }?.roundToInt()
-    }
-
-  fun getWeightInKilograms() =
-    // Take weight in lb and convert if it exists because the data is more accurate (being a smaller unit than kg). See the unit tests for an example explaining why.
-    if (weightPounds != null) {
-      weightPounds!!.let { (it * 0.453592) }.roundToInt()
-    } else {
-      weightKilograms
-    }
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/OffenderPhysicalAttributes.kt:84]: Moved to the service layer - this logic only makes sense in the context of migration / sync.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonIntTest.kt [hunk 2]
```diff
@@ -319,6 +331,116 @@ class PrisonPersonIntTest : IntegrationTestBase() {
             }
           }
       }
+
+      @Test
+      fun `should return null if empty attributes`() {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonIntTest.kt:336]: Imperial measures conversion tests moved here.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt [hunk 3]
```diff
@@ -57,4 +63,47 @@ class PrisonPersonService(
...
+    val booking = bookingRepository.findLatestByOffenderNomsId(offenderNo)
+      ?: throw NotFoundException("No latest booking found for $offenderNo")
+
+    val physicalAttributes = booking.physicalAttributes.find { it.id.sequence == 1L }
+      ?: OffenderPhysicalAttributes(id = OffenderPhysicalAttributeId(booking, 1L))
+
+    physicalAttributes.setWeightInKilograms(request.weight)
+    physicalAttributes.setHeightInCentimetres(request.height)
+    offenderPhysicalAttributesRepository.save(physicalAttributes)
+  }
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt:77]: The main change is to implement this service.

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt [hunk 4]
```diff
@@ -57,4 +63,47 @@ class PrisonPersonService(
...
+  fun upsertPhysicalAttributes(offenderNo: String, request: UpsertPhysicalAttributesRequest) {
+    val booking = bookingRepository.findLatestByOffenderNomsId(offenderNo)
+      ?: throw NotFoundException("No latest booking found for $offenderNo")
+
+    val physicalAttributes = booking.physicalAttributes.find { it.id.sequence == 1L }
+      ?: OffenderPhysicalAttributes(id = OffenderPhysicalAttributeId(booking, 1L))
+
+    physicalAttributes.setWeightInKilograms(request.weight)
+    physicalAttributes.setHeightInCentimetres(request.height)
+    offenderPhysicalAttributesRepository.save(physicalAttributes)
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt:76]: I guess you don't need the `save` given you are amending the booking that was just retrieved ?

Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt:76]: It could be a new record - see line 72.

**Pull request merged**